### PR TITLE
Speed up DB tests

### PR DESCRIPTION
**Pull request by camdencheek**
With the perf improvements from #21300, `dbtest.NewDB()` is faster than `dbtesting.GetDB()`, and allows tests to be run in parallel. 

This change converts (almost) all instances of `dbtesting.GetDB()` in `internal/database` to `dbtest.NewDB()`, and sets `t.Parallel()` for tests that don't rely on any global state. 

Additionally, since tests are being run in a more parallel fashion, we were hitting the "too many connections" error more often. I added a max open connections limit of 3 to all database connections created with `dbtest.NewDB()` which seems to have helped. 

This change reduced runtime for `internal/database` to ~50s (from ~80 seconds in #21300, which was already reduced from ~120 seconds). 

```
❯ for i in {1..20} ; do go clean -testcache && go test ./internal/database ; done
ok      github.com/sourcegraph/sourcegraph/internal/database    33.562s
ok      github.com/sourcegraph/sourcegraph/internal/database    51.293s
ok      github.com/sourcegraph/sourcegraph/internal/database    51.237s
ok      github.com/sourcegraph/sourcegraph/internal/database    50.243s
ok      github.com/sourcegraph/sourcegraph/internal/database    48.176s
ok      github.com/sourcegraph/sourcegraph/internal/database    49.913s
ok      github.com/sourcegraph/sourcegraph/internal/database    50.085s
ok      github.com/sourcegraph/sourcegraph/internal/database    50.945s
ok      github.com/sourcegraph/sourcegraph/internal/database    48.659s
ok      github.com/sourcegraph/sourcegraph/internal/database    51.025s
```

### CODE REVIEW

**Comment by camdencheek** [comment]
@mrnugget To answer your question about whether `dbtest.NewDB()` is now faster than `dbtesting.GetDB()`, I think this PR is evidence that it is

File: internal/database/event_logs_test.go [hunk 1]
```diff
@@ -112,7 +115,8 @@ func TestEventLogs_UsersUsageCounts(t *testing.T) {
 	if testing.Short() {
 		t.Skip()
 	}
-	db := dbtesting.GetDB(t)
+	t.Parallel()
+	db := dbtest.NewDB(t, "")
```
Comment (unknwon) [inline internal/database/event_logs_test.go:119]: I have a thoughtful thought, can we just call `t.Parallel()` within `dbtest.NewDB(t, "")`? Would that work?

**Review by unknwon — APPROVED** [review]

File: internal/database/event_logs_test.go (hunk 1 above)
Comment (camdencheek) [reply internal/database/event_logs_test.go:119]: My concern with that is hiding that the test is running in parallel. In tests that use global state (global mock overrides, etc), they can't be run in parallel, but as it is now, they can still use dbtest.NewDB()

Comment (unknwon) [reply internal/database/event_logs_test.go:119]: Right right, better to be more explicit 😁

**Comment by camdencheek** [comment]
A little more information on where I'm leaving performance:

On average, a call to `NewDB()` takes 1.25 seconds to return, and adds an additional 0.6 seconds to the test cleanup phase. 

When the tests are run with `-parallel 1`, this changes significantly to 0.31 seconds to create the DB and 0.14 seconds to clean it up once the test completes. (note that it still takes ~30 seconds longer overall to run the tests with no parallel)

These two things together indicate to me that much of the difference is due to contention on the Postgres system catalog from all the create/drop operations. If that's the case, there is still likely some improvement to performance on the table if we maintain a pool of migrated databases and just truncate all the tables before returning them. I don't plan to pursue this further at the moment, but figured it would be interesting context to add to the PR.

**Review by eseliger — APPROVED** [review]
n i c e

**Review by asdine — APPROVED** [review]
👏🏼

**Review by mrnugget — APPROVED** [review]
NICE!

**Pull request merged**